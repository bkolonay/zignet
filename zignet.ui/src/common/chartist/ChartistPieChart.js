import React, { Component } from 'react';
import Chartist from 'chartist';
import './css/chartist.css';
import './css/chartistPieChart.css';

class ChartistPieChart extends Component {

  componentDidMount() {
  	this.chartistChart = Chartist.Pie(
  		this.chartContainer,
  		this.props.chartData,
  		{ 
  			donut: true,
			  donutWidth: 40
		  }
	  )
  }

  componentWillReceiveProps(props) {
  	this.chartistChart.update(props.chartData);
  }

  render() {
    return (
      <div 
        className="ct-chart" 
        id={"chart" + this.props.chartId}
        ref={(div) => this.chartContainer = div}
      />
    );
  }
}

export default ChartistPieChart;
