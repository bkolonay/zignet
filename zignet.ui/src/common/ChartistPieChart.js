import React, { Component } from 'react';
import Chartist from 'chartist';
import './chartist.css';

class ChartistPieChart extends Component {

  componentDidMount() {
  	this.chartistChart = Chartist.Pie(
  		this.chartContainer,
  		this.props.chartData,
  		{ 
  			donut: true,
			donutWidth: 40
		}
	)
  }

  render() {
    return (
      <div 
        className="ct-chart" 
        id={"chart" + this.props.chartId}
        ref={(div) => this.chartContainer = div}
      />
    );
  }
}

export default ChartistPieChart;
