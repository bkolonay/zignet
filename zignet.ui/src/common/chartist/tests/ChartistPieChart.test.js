import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import ChartistPieChart from '../ChartistPieChart';

it('renders with valid data', () => {
  const component = <ChartistPieChart chartId={1} chartData={{ series: [123, 456] }} />
  ReactDOM.render(component, document.createElement('div'));
});

it('renders with empty data', () => {
  const component = <ChartistPieChart />
  ReactDOM.render(component, document.createElement('div'));
});

it('snapshot with valid data', () => {
  const component = <ChartistPieChart chartId={123} chartData={{ series: [123, 456] }} />
  const tree = renderer.create(component).toJSON();  
  expect(tree).toMatchSnapshot();
});

it('snapshot with no data', () => {
  const component = <ChartistPieChart />
  const tree = renderer.create(component).toJSON();  
  expect(tree).toMatchSnapshot();
});