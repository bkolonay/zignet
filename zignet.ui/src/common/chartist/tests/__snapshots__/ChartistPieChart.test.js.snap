// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the previous snapshot 1`] = `
<div
  className="ct-chart"
  id="chart321"
/>
`;
